### PR TITLE
Conform to PEP 561

### LINKED ISSUE #27: Add missing typed marker
When running mypy on my code that I have now have type hinted I get the following
image
Can this be added to fix this error please?

### PR DESCRIPTION
**Pull request by DavidPratt512**
Refactor the project to have a PEP 561 compliant directory structure. This will allow other dependent packages to use mypy without any hiccups. 

Closes #27

### CODE REVIEW

File: wakeonlan/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from .wakeonlan import BROADCAST_IP
+from .wakeonlan import DEFAULT_PORT
+from .wakeonlan import create_magic_packet
+from .wakeonlan import send_magic_packet
+from .wakeonlan import main
```
Comment (remcohaszing) [inline wakeonlan/__init__.py:5]: Should we expose the main function? I’m leaning towards no. That would mean the `wakeonlan` script in `pyproject.toml` needs to be updated.

**Review by remcohaszing — CHANGES_REQUESTED** [review]
Thanks! Looks great!

I fixed CI. Could you rebase?

File: wakeonlan/__init__.py (hunk 1 above)
Comment (DavidPratt512) [reply wakeonlan/__init__.py:5]: I'm also leaning towards no. I kept it here just to make sure not to break compatibility. Can't imagine anyone is calling `main()` from their script though.

Haven't been in a python environment for a while, but I think the idiomatic thing to do would be:

- rename to `_main`
- remove the `_main` import from `__init__.py`
- call `_main` from `__main__.py`
- repoint the script property in `pyproject.toml` to `__main__` (I think...)

Thoughts on that?

File: wakeonlan/__init__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from .wakeonlan import BROADCAST_IP
```
Comment (remcohaszing) [inline wakeonlan/__init__.py:1]: We should also expose `__doc__`.

File: wakeonlan/__init__.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+from .wakeonlan import send_magic_packet
+
+
+__all__ = (
+    "BROADCAST_IP",
+    "DEFAULT_PORT",
+    "create_magic_packet",
+    "send_magic_packet",
+    "main",
+)
```
Comment (remcohaszing) [inline wakeonlan/__init__.py:14]: Do we need this? I think we either need to manually cherry-pick the imports above, or use `from .wakeonlan import *` and define `__all__` in `wakeonlan.py`, but not both.

**Review by remcohaszing — CHANGES_REQUESTED** [review]

File: wakeonlan/__init__.py (hunk 1 above)
Comment (remcohaszing) [reply wakeonlan/__init__.py:4]: I had to drop Python 3.6 support in order to get pytest working in CI, so this is already going to be a semver major release.

It appears `__main__.py` is used by [`zipapp`](https://docs.python.org/3/library/zipapp.html) to create executable zips. It looks nice, but also redundant for packages that consist of a single file. If we just keep `wakeonlan.py` as-is, people can use that instead.

Let’s just remove the `main` export here. :)

**Comment by remcohaszing** [comment]
I think those `@patch` decoratores in tests need to be changed from `wakeonlan.…` to `wakeonlan.wakeonlan.…` to fix the tests.

**Comment by DavidPratt512** [comment]
So I decided to just move everything to `__init__.py`. That gets us:

- exposes `__doc__`
- no need to change tests
- don't need to screw around with `__all__`

The only downside is that the `main` function is still exposed. But that's an easy change.

**Comment by remcohaszing** [comment]
Thank you!

**Comment by remcohaszing** [comment]
This has been released in version 3.0.0.

**Pull request merged**